core/unix/loader.c

```diff
@@ -741,17 +747,48 @@ privload_search_rpath(privmod_t *mod, bool runpath, const char *name,
                     len = sep - list;
                 /* support $ORIGIN expansion to lib's current directory */
                 origin = strstr(list, RPATH_ORIGIN);
+                char path[MAXIMUM_PATH];
                 if (origin != NULL && origin < list + len) {
                     size_t pre_len = origin - list;
-                    snprintf(filename, MAXIMUM_PATH, "%.*s%.*s%.*s/%s", pre_len, list,
+                    snprintf(path, BUFFER_SIZE_BYTES(path), "%.*s%.*s%.*s", pre_len, list,
```

Comment:
bug: BUFFER_SIZE_ELEMENTS (bug if ever converted to wchar_t on Windows or sthg -- never use _BYTES w/ print routines: always elements)

Reply:
Should have verified when doing [this](https://github.com/DynamoRIO/dynamorio/pull/3854#discussion_r328911669)

Fixed.